### PR TITLE
switch to the smoke testing included in heffte

### PR DESCRIPTION
**Pull request by mkstoyanov**
* switch to the smoke testing included in heffte

### CODE REVIEW

File: var/spack/repos/builtin/packages/heffte/package.py [hunk 1]
```diff
@@ -97,59 +97,10 @@ def cmake_args(self):
...
-                cmkf.write(
-                    'target_link_libraries(hf_fort Heffte::Fortran)\n')
-                exe_files.append('./hf_fort')
-                mpi_procs.append(2)
-
-        with working_dir(test_bld_dir, create=True):
-            cmake('..')
+        # using the tests installed in <prefix>/share/heffte/testing
+        cmake_dir = join_path(self.prefix, 'share', 'heffte', 'testing')
+        with working_dir(self.install_test_root, create=True):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/heffte/package.py]: Building within any install root directory has permissions issues with facility deployments and shared Spack instances.

Is it at all possible to do this from`self.test_suite.current_test_cach_dir`?  You would need to create the directory -- or an alternate subdirectory of it if you are no longer using `cache_test_sources`.  For example:

```suggestion
        with working_dir(self.test_suite.current_test_cache_dir, create=True):
```

If you have issues with the directory we may want to add a new property that gives ready access to a suitable alternative.

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Comment by tldahlgren** [comment]
:heart: Much better to (re-)use code maintained in the software repository so the tests can evolve with the code base!

**Comment by tldahlgren** [comment]
I also prefer to see the use of `self.run_test()` so execution of each step is logged to a test log but, since the calls are dependent, it is not strictly necessary.

**Comment by tldahlgren** [comment]
The pipeline build is failing due to `phist` (see #26312).

File: var/spack/repos/builtin/packages/heffte/package.py (hunk 1 above)
Comment (mkstoyanov) [reply var/spack/repos/builtin/packages/heffte/package.py]: I changed the folder to a sub-folder of `self.test_suite.current_test_cache_dir`

All the test needs is a clean folder to run CMake.

**Comment by mkstoyanov** [comment]
> I also prefer to see the use of `self.run_test()` so execution of each step is logged to a test log but, since the calls are dependent, it is not strictly necessary.

The `make('test'')` step uses ctest which provides individual logging of the results. In case of an error, spack reports the correct log file to check. Using `run_test()` for each individual executable will just mirror functionality already existing in CMake.

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**